eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/ResourceProvider.cs

```diff
@@ -0,0 +1,345 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Generator.Mgmt.Models;
+using Azure.Generator.Primitives;
+using Azure.Generator.Utilities;
+using Azure.ResourceManager;
+using Microsoft.Generator.CSharp.ClientModel.Providers;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Providers;
+using Microsoft.Generator.CSharp.Snippets;
+using Microsoft.Generator.CSharp.Statements;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using static Microsoft.Generator.CSharp.Snippets.Snippet;
+
+namespace Azure.Generator.Providers
+{
+    internal class ResourceProvider : TypeProvider
+    {
+        private OperationSet _operationSet;
+        private ResourceDataProvider _resourceData;
+        private ClientProvider _clientProvider;
+        private string _specCleanName;
+        private readonly HashSet<string> _contextualParameters;
+
+        private FieldProvider _dataField;
+        private FieldProvider _clientDiagonosticsField;
+        private FieldProvider _restClientField;
+        private FieldProvider _resourcetypeField;
+
+        public ResourceProvider(OperationSet operationSet, string specCleanName, ResourceDataProvider resourceData, string resrouceType)
+        {
+            _operationSet = operationSet;
+            _specCleanName = specCleanName;
+            _resourceData = resourceData;
+            _clientProvider = AzureClientPlugin.Instance.TypeFactory.CreateClient(operationSet.InputClient)!;
```

Comment:
so now we will be generate:
- a resource class, its operations call the operations in `_clientProvider`
- a XXXClient class, this works as our `XXXRestOperations`
- a XXXClient.RestClient (partial class of above), it contains the building request, etc.
therefore as a result, we still have two classes, but 3 files
is this correct?

Reply:
Not quite following your concern here.
XXXClient file and XXXClient.RestClient file are generated by MGC, are you saying if should be a single file instead?

Reply:
I do not have concerns here just confirming how the generated code would look like comparing with the code we used to have.

Reply:
Offline discussed with @m-nash about the code structure, conclusion is https://github.com/Azure/azure-sdk-for-net/pull/47944#discussion_r1971420479